test/CMakeLists.txt

```diff
@@ -47,12 +47,14 @@ add_executable(
   test_eos_stellar_collapse.cpp
   )
 
-add_executable(
-  closure_unit_tests
-  catch2_define.cpp
-  test_closure_pte.cpp
-  test_kpt_models.cpp
-  )
+if (SINGULARITY_BUILD_CLOSURE)
+  add_executable(
+    closure_unit_tests
+    catch2_define.cpp
+    test_closure_pte.cpp
+    test_kpt_models.cpp
+    )
+endif()
```

Comment:
When I added the closure unit tests, I was copying the approach for the tabulated unit tests and trying to simplify the CMake so that each option gets applied to its relevant executables one-by-one. Is it better to complicate the CMake logic to avoid compiling the test rather than rely on `#ifdef` statements to make sure its basically an empty test?

Put another way, in general should we be trying to push more logic into the CMake and less into the source via preprocessor flags?

Reply:
Oh I don't know. Either is fine. I went overboard here, the tests weren't wrapped in preprocessor protection in this case or protected in the cmake so I just did both to be safe.

Reply:
I'm pretty ignorant to the finer points of CMake so I'm happy to go either way. But if there's a superior way, let me know. @mauneyc-LANL do you have opinions?

Reply:
👍 Yeah happy to revise this structure if there's one that makes sense.